clients/python/agentic-sandbox-client/agentic_sandbox/sandbox_client.py

```diff
@@ -113,37 +121,50 @@ def _wait_for_sandbox_ready(self):
             field_selector=f"metadata.name={self.claim_name}",
             timeout_seconds=self.sandbox_ready_timeout
         ):
-            sandbox_object = event['object']
-            status = sandbox_object.get('status', {})
-            conditions = status.get('conditions', [])
-            is_ready = False
-            for cond in conditions:
-                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
-                    is_ready = True
-                    break
+            if event["type"] in ["ADDED", "MODIFIED"]:
+                sandbox_object = event['object']
+                status = sandbox_object.get('status', {})
+                conditions = status.get('conditions', [])
+                is_ready = False
+                for cond in conditions:
+                    if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                        is_ready = True
+                        break
 
-            if is_ready:
-                self.sandbox_name = sandbox_object['metadata']['name']
-                w.stop()
-                logging.info(f"Sandbox {self.sandbox_name} is ready.")
-                break
+                if is_ready:
+                    self.sandbox_name = sandbox_object['metadata']['name']
+                    annotations = sandbox_object.get(
+                        'metadata', {}).get('annotations', {})
+                    pod_name_annotation = "agents.x-k8s.io/pod-name"
```

Comment:
The annotation key `"agents.x-k8s.io/pod-name"` is hardcoded. This can lead to typos and inconsistencies if it's used elsewhere. Please define this as a constant at the module level, for example: `POD_NAME_ANNOTATION = "agents.x-k8s.io/pod-name"`.